### PR TITLE
Prune on `initial_routing_sync`

### PR DESCRIPTION
**Pull request by pm47**
This PR brings several improvements:
- announcements are pruned before they are sent for `initial_routing_sync` (this was removed by #422)
- is now considered stale a channel that is older than 2 weeks and hasn't any `channel_update` younger than two weeks (again, before #422 we were more lax because of pre-validation pruning)
- pruning was rewritten without use of very expensive `exists` function
- we don't prune anymore on startup
- removed `ThrottleForwarder` and `WriteAckSender`, and handle both TCP read/write backpressure at the `TransportHandler` level, which is much more efficient (e.g. we don't serialize/encrypt outgoing data if OS buffer is already full, which saves RAM and CPU)
- added a simple priority management that reduces priority of routing announcements so that they don't block channel messages on `initial_routing_sync`

### CODE REVIEW

File: eclair-core/src/main/scala/fr/acinq/eclair/crypto/TransportHandler.scala [hunk 1]
```diff
@@ -109,60 +108,88 @@ class TransportHandler[T: ClassTag](keyPair: KeyPair, rs: Option[BinaryData], co
...
+        log.warning(s"send buffer overrun, closing connection")
+        connection ! PoisonPill
+        stop(FSM.Normal)
+      } else if (d.unackedSent.isDefined) {
+        log.debug("buffering send data={}", t)
+        val sendBuffer1 = t match {
+          case _: ChannelAnnouncement => d.sendBuffer.copy(lowPriority = d.sendBuffer.lowPriority :+ t)
+          case _: NodeAnnouncement => d.sendBuffer.copy(lowPriority = d.sendBuffer.lowPriority :+ t)
+          case _: ChannelUpdate => d.sendBuffer.copy(lowPriority = d.sendBuffer.lowPriority :+ t)
+          case _ => d.sendBuffer.copy(lowPriority = d.sendBuffer.normalPriority :+ t)
```
Comment (sstone) [inline eclair-core/src/main/scala/fr/acinq/eclair/crypto/TransportHandler.scala]: It should be ```d.sendBuffer.copy(normalPriority = d.sendBuffer.normalPriority :+ t)``` ?

File: eclair-core/src/main/scala/fr/acinq/eclair/crypto/TransportHandler.scala [hunk 2]
```diff
@@ -109,60 +108,88 @@ class TransportHandler[T: ClassTag](keyPair: KeyPair, rs: Option[BinaryData], co
...
-      stay using d.copy(enc = enc1)
+    case Event(WriteAck, d: NormalData[T]) =>
+      def send(t: T) = {
+        val blob = codec.encode(t).require.toByteArray
+        val (enc1, ciphertext) = d.encryptor.encrypt(blob)
+        connection ! Tcp.Write(buf(ciphertext), WriteAck)
+        enc1
+      }
+      d.sendBuffer.normalPriority.dequeueOption match {
+        case Some((t, highPriority1)) =>
```
Comment (sstone) [inline eclair-core/src/main/scala/fr/acinq/eclair/crypto/TransportHandler.scala]: nitpick: why not call is normalPriority1 ?

File: eclair-core/src/main/scala/fr/acinq/eclair/crypto/TransportHandler.scala (hunk 1 above)
Comment (pm47) [reply eclair-core/src/main/scala/fr/acinq/eclair/crypto/TransportHandler.scala]: great catch :+1:

File: eclair-core/src/main/scala/fr/acinq/eclair/crypto/TransportHandler.scala (hunk 2 above)
Comment (pm47) [reply eclair-core/src/main/scala/fr/acinq/eclair/crypto/TransportHandler.scala]: that's a leftover, will fix

**Pull request merged**